test/methodsSpec.js

```diff
@@ -37,47 +37,63 @@ describe('Methods: ', function() {
 		done();
 	});
 
-	describe("Asynchronous specs:", function() {
+	describe("Asynchronous specs:", function () {
 		var originalTimeout;
-		beforeEach(function() {
+		beforeEach(function () {
 			originalTimeout = jasmine.DEFAULT_TIMEOUT_INTERVAL;
 			jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000;
 		});
 
-		it("Pause method is working", function(done) {
-			setTimeout(function() {
+		it("Pause method is working", function (done) {
+			setTimeout(function () {
 				granimInstance.pause();
 				expect(granimInstance.isPaused).toBe(true);
 				done();
 			}, 105);
 		});
 
-		it("Play method is working", function(done) {
-			setTimeout(function() {
+		it("Play method is working", function (done) {
+			setTimeout(function () {
 				granimInstance.play();
 				expect(granimInstance.isPaused).toBe(false);
 				done();
 			}, 10);
 		});
 
-		it("ChangeState method is working", function(done) {
-			setTimeout(function() {
+		it("ChangeState method is working", function (done) {
+			setTimeout(function () {
 				granimInstance.changeState('default-state');
 				granimInstance.changeState('second-state');
 				expect(granimInstance.activeState).toEqual('second-state');
 				done();
 			}, 200);
 		});
 
-		it("Clear method is working", function(done) {
-			setTimeout(function() {
+		it("ChangeDirection method is working", function (done) {
+			setTimeout(function () {
+				granimInstance.changeDirection('left-right');
+				expect(granimInstance.direction).toEqual('left-right');
+				done();
+			}, 200);
+		});
+
+		it("Clear method is working", function (done) {
+			setTimeout(function () {
 				granimInstance.clear();
 				expect(granimInstance.context.getImageData(1, 1, 1, 1).data[0]).toEqual(0);
 				done();
 			}, 10);
 		});
 
-		afterEach(function() {
+		it("ChangeDirection should throw an error when passed 'custom' value without a customDirection object", function (done) {
+			setTimeout(function () {
+				var func = function () { return granimInstance.changeDirection('custom'); };
+				expect(func).toThrowError('When using a custom direction, the custom object is required with it\'s attributes: x0, x1, y0, y1 of type number');
```

Comment:
`expect(func).toThrowError('Granim: Input error on customDirection option.\nCheck the API https://sarcadass.github.io/granim.js/api.html.');`